malformed yml causes panic

### Summary

Opening a yaml file beginning with ']' causes helix to panic:

```
thread 'main' panicked at 'Attempt to index past end of slice: char index 0, slice char length 0', /home/jake/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ropey-1.6.1/src/slice.rs:492:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Reproduction Steps

```sh
echo ']' > broken.yml
hx broken.yml
```

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2024-01-13T00:29:32.643 helix_view::clipboard [DEBUG] Using xclip to interact with the system and selection (primary) clipboard
2024-01-13T00:29:32.702 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/home/jake",
        },
    ),
}
2024-01-13T00:29:32.702 helix_vcs [INFO] failed to open diff base for /home/jake/broken.yml
2024-01-13T00:29:32.702 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/home/jake",
        },
    ),
}
2024-01-13T00:29:32.702 helix_vcs [INFO] failed to obtain current head name for /home/jake/broken.yml
2024-01-13T00:29:32.702 helix_view::editor [ERROR] Failed to initialize the language servers for `source.yaml` - `yaml-language-server` { cannot find binary path }
2024-01-13T00:29:32.703 helix_view::editor [ERROR] Failed to initialize the language servers for `source.yaml` - `ansible-language-server` { cannot find binary path }
2024-01-13T00:29:32.703 helix_view::editor [DEBUG] editor status: Loaded 1 file.
2024-01-13T00:29:32.704 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is supported in this terminal (checked in 286.738µs)
```

</details>


### Platform

Linux

### Terminal Emulator

alacritty 0.13.1 (fe2a3c56)

### Installation Method

aur/helix-git

### Helix Version

helix 23.10 (17dd102e)

--- a/helix-core/src/match_brackets.rs
+++ b/helix-core/src/match_brackets.rs
@@ -141,7 +141,7 @@ fn find_pair(
 #[must_use]
 pub fn find_matching_bracket_plaintext(doc: RopeSlice, cursor_pos: usize) -> Option<usize> {
     // Don't do anything when the cursor is not on top of a bracket.
-    let bracket = doc.char(cursor_pos);
+    let bracket = doc.get_char(cursor_pos)?;
     if !is_valid_bracket(bracket) {
         return None;
     }
@@ -264,6 +264,12 @@ fn as_char(doc: RopeSlice, node: &Node) -> Option<(usize, char)> {
 #[cfg(test)]
 mod tests {
     use super::*;
+
+    #[test]
+    fn find_matching_bracket_empty_file() {
+        let actual = find_matching_bracket_plaintext("".into(), 0);
+        assert_eq!(actual, None);
+    }
 
     #[test]
     fn test_find_matching_bracket_current_line_plaintext() {

--- a/helix-core/src/match_brackets.rs
+++ b/helix-core/src/match_brackets.rs
@@ -60,7 +60,7 @@ fn find_pair(
     let tree = syntax.tree();
     let pos = doc.char_to_byte(pos_);
 
-    let mut node = tree.root_node().descendant_for_byte_range(pos, pos)?;
+    let mut node = tree.root_node().descendant_for_byte_range(pos, pos + 1)?;
 
     loop {
         if node.is_named() {
@@ -118,7 +118,9 @@ fn find_pair(
         };
         node = parent;
     }
-    let node = tree.root_node().named_descendant_for_byte_range(pos, pos)?;
+    let node = tree
+        .root_node()
+        .named_descendant_for_byte_range(pos, pos + 1)?;
     if node.child_count() != 0 {
         return None;
     }

